### DISCUSSION TITLE
dotenv in azure DevOps

### DISCUSSION THREAD

**Discussion by abdurraheemj** [seleniumbase/SeleniumBase#1232] [Q&A] [answered]
Hi Michael,
I have my script which uses username and password. I am using .env file for that. It works perfectly fine in local. when i go for Azure DevOps its not reading the value from .env file . FYI: I stored my .env file in Azure DevOps
Error Message: AttributeError: 'NoneType' object has no attribute 'endswith' . Please help me

**Comment by mdmintz** [comment, marked as answer]
Not sure what’s going on without a full stack trace, and that doesn’t sound related to SeleniumBase, but “.endswith” must be called on a string, and it appears that isn’t the case.
